modules/highlight/highlight_test.go

```diff
@@ -8,100 +8,116 @@ import (
+
 	tests := []struct {
-		name     string
-		numLines int
-		fileName string
-		code     string
-		want     string
+		name string
+		code string
+		want []string
 	}{
 		{
-			name:     ".drone.yml",
-			numLines: 12,
-			fileName: ".drone.yml",
-			code: util.Dedent(`
-				kind: pipeline
-				name: default
+			name: "empty.py",
+			code: "",
+			want: []string{""},
+		},
+		{
+			name: "eol-no.py",
+			code: "a=1",
+			want: []string{`<span class="n">a</span><span class="o">=</span><span class="mi">1</span>`},
+		},
+		{
+			name: "eol-newline1.py",
+			code: "a=1\n",
+			want: []string{
+				`<span class="n">a</span><span class="o">=</span><span class="mi">1</span>&#10;`,
+			},
+		},
+		{
+			name: "eol-newline2.py",
+			code: "a=1\n\n",
+			want: []string{
+				`<span class="n">a</span><span class="o">=</span><span class="mi">1</span>&#10;`,
+				`&#10;`,
+			},
+		},
+		{
+			name: "empty-line.py",
+			code: strings.TrimSpace(`
+a=1
 
-				steps:
-				- name: test
-					image: golang:1.13
-					environment:
-						GOPROXY: https://goproxy.cn
-					commands:
-					- go get -u
-					- go build -v
-					- go test -v -race -coverprofile=coverage.txt -covermode=atomic
+b=''
 			`),
-			want: util.Dedent(`
-				<span class="nt">kind</span><span class="p">:</span><span class="w"> </span><span class="l">pipeline</span><span class="w">
-				</span><span class="w"></span><span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">default</span><span class="w">
-				</span><span class="w">
+			want: []string{
+				`<span class="n">a</span><span class="o">=</span><span class="mi">1</span>&#10;`,
+				`&#10;`,
+				`<span class="n">b</span><span class="o">=</span><span class="sa"></span><span class="s1">&#39;</span><span class="s1">&#39;</span>`,
+			},
+		},
+	}
 
-				</span><span class="w"></span><span class="nt">steps</span><span class="p">:</span><span class="w">
-				</span><span class="w"></span>- <span class="nt">name</span><span class="p">:</span><span class="w"> </span><span class="l">test</span><span class="w">
-				</span><span class="w">	</span><span class="nt">image</span><span class="p">:</span><span class="w"> </span><span class="l">golang:1.13</span><span class="w">
-				</span><span class="w">	</span><span class="nt">environment</span><span class="p">:</span><span class="w">
-				</span><span class="w"></span><span class="w">		</span><span class="nt">GOPROXY</span><span class="p">:</span><span class="w"> </span><span class="l">https://goproxy.cn</span><span class="w">
-				</span><span class="w">	</span><span class="nt">commands</span><span class="p">:</span><span class="w">
-				</span><span class="w"></span><span class="w">	</span>- <span class="l">go get -u</span><span class="w">
-				</span><span class="w">	</span>- <span class="l">go build -v</span><span class="w">
-				</span><span class="w">	</span>- <span class="l">go test -v -race -coverprofile=coverage.txt -covermode=atomic</span>
-			`),
```

Comment:
Why remove these test and not just extend with new cases?

Reply:
Does these old tests are for a special case? If not, I prefer to keep the tests more simple and more clear.

I do not think using complex data for unit tests are easy to be maintained.

Reply:
I guess if the new tests include all aspects of the previous one, it should be fine.